shared/markdown/parser.pegjs

```diff
@@ -117,14 +117,20 @@ __INLINE_RULE__<StrikeInline, !StrikeMarker>
 Strike
  = StrikeMarker !WhiteSpace children:StrikeInline StrikeMarker !(StrikeMarker / NormalChar) { return {type: 'strike', children: flatten(children)} }
 
-// children grammar adapted from username regexp in libkb/checkers.go.
+// children test adapted from CheckUsername in libkb/checkers.go.
 Mention
- = MentionMarker children:([a-zA-Z0-9][a-zA-Z0-9_]*) MentionMarker service:ValidMentionService { return {type: 'mention', children: flatten(children), service: service.toLowerCase()} }
```

Comment:
The problem here with the old grammar `([a-zA-Z0-9][a-zA-Z0-9_]?)+` is that this isn't a regexp, but a grammar, and so would parse greedily and not backtrack. e.g. `m_` and `dan_t` are okay because the underscore falls on an odd index, but `ryan_singer` isn't!